src/commands.cpp

```diff
@@ -2173,146 +2325,240 @@ void CommandPutUA::procresult()
     }
     else
     {
-        client->json.storeobject();
+        client->json.storeobject(); // [<uh>]
         e = API_OK;
+
+        User *u = client->ownuser();
+        u->setattr(at, &av, NULL);
+        u->setTag(tag ? tag : -1);
+
+        client->notifyuser(u);
     }
 
     client->app->putua_result(e);
 }
 
-CommandGetUA::CommandGetUA(MegaClient* client, const char* uid, const char* an)
+CommandGetUA::CommandGetUA(MegaClient* client, const char* uid, attr_t at, int ctag)
 {
-    user = client->finduser((char*)uid);
-    attributename = an;
+    this->u = client->finduser((char*)uid);
```

Comment:
It's not specific of this pull request and there are other commands doing the same, but it's potentially dangerous to save this pointer. A fetchnodes in the same request before this command would invalidate the pointer.